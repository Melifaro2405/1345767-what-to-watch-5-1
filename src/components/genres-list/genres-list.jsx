import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {changeActiveFilter} from "../../store/action";

const GenresList = ({genres, activeGenre, changeActiveFilterAction}) => {
  return (
    <ul className="catalog__genres-list">
      {genres.map((genre, index) =>
        <li key={`genre-${index}`} onClick={(evt)=>{
          evt.preventDefault();
          changeActiveFilterAction(genre);
        }} className={`catalog__genres-item ${activeGenre === genre ? `catalog__genres-item--active` : ``}`}>
          <a href="#" className="catalog__genres-link">{genre}</a>
        </li>)}
    </ul>
  );
};

GenresList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  activeGenre: PropTypes.string.isRequired,
  changeActiveFilterAction: PropTypes.func.isRequired,
};

const mapStateToProps = ({DATA, APP_STATE}) => ({
  genres: DATA.genres,
  activeGenre: APP_STATE.activeGenre
});

const mapDispatchToProps = (dispatch) => ({
  changeActiveFilterAction(genre) {
    dispatch(changeActiveFilter(genre));
  },
});

export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
