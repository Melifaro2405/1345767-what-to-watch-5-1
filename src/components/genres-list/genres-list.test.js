import React from 'react';
import renderer from 'react-test-renderer';
import {GenresList} from './genres-list';

const noop = () => {};

describe(`Should GenresList render correctly`, () => {
  test.each([
    [`with`, `test`],
    [`without`, ``],
  ])(`%s activeGenre`, (_expected, activeGenre) => {
    const tree = renderer
      .create(<GenresList
        genres={[`test1, test2, test3`]}
        activeGenre={activeGenre}
        changeActiveFilterAction={noop}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
