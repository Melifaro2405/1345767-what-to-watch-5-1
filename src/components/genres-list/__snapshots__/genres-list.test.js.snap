// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenresList render correctly with activeGenre 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      test1, test2, test3
    </a>
  </li>
</ul>
`;

exports[`Should GenresList render correctly without activeGenre 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      test1, test2, test3
    </a>
  </li>
</ul>
`;
